source/Mlos.Python/mlos/Optimizers/OptimumDefinition.py

```diff
@@ -43,6 +43,8 @@ class OptimumDefinition(Enum):
     #
     LOWER_CONFIDENCE_BOUND_FOR_OBSERVED_CONFIG = 'prediction_lower_confidence_bound_for_observed_config'
 
+    BEST_PREDICTED_WITHIN_CONTEXT = 'best_predicted_within_context'

```

Comment:
For now I left it as-is to keep the PR small-ish. We can add a different way to specify the different optima later maybe?

Reply:
Sure, but please rename it to BEST_SPECULATIVE_WITHIN_CONTEXT or something like that. This is potentially very risky and the user should be made aware.

Reply:
done :)